.github/workflows/nfr.yml

```diff
@@ -0,0 +1,196 @@
+
+    - name: Build binary
+      uses: goreleaser/goreleaser-action@7ec5c2b0c6cdda6e8bbb49444bc797dd33d74dd8 # v5.0.0
+      with:
+        version: latest
+        args: build --snapshot --clean
+
+    - name: Authenticate to Google Cloud
+      id: auth
+      uses: google-github-actions/auth@a6e2e39c0a0331da29f7fd2c2a20a427e8d3ad1f # v2.1.1
+      with:
+        token_format: access_token
+        workload_identity_provider: ${{ secrets.GCP_WORKLOAD_IDENTITY }}
+        service_account: ${{ secrets.GCP_SERVICE_ACCOUNT }}
+
+    - name: Set up Cloud SDK
+      uses: google-github-actions/setup-gcloud@98ddc00a17442e89a24bbf282954a3b65ce6d200 # v2.1.0
+      with:
+        project_id: '${{ secrets.GCP_PROJECT_ID }}'
+        install_components: 'kubectl'
+
+    - name: Login to GCR
+      run: gcloud auth configure-docker gcr.io -q
+
+    - name: Build NGF Docker Image
+      uses: docker/build-push-action@4a13e500e55cf31b7a5d59a38ab2040ab0f42f56 # v5.1.0
+      with:
+        file: build/Dockerfile
+        tags: ${{ steps.ngf-meta.outputs.tags }}
+        context: "."
+        provenance: false
+        platforms: linux/amd64
+        target: goreleaser
+        pull: true
+        push: true
+
+    - name: Build NGINX Docker Image
+      uses: docker/build-push-action@4a13e500e55cf31b7a5d59a38ab2040ab0f42f56 # v5.1.0
+      with:
+        file: build/Dockerfile${{ inputs.nginx-plus == true && '.nginxplus' || '.nginx'}}
+        tags: ${{ steps.nginx-meta.outputs.tags }}
+        context: "."
+        platforms: linux/amd64
+        provenance: false
+        pull: true
+        push: true
+        build-args: |
+          NJS_DIR=internal/mode/static/nginx/modules/src
+          NGINX_CONF_DIR=internal/mode/static/nginx/conf
+          BUILD_AGENT=gha
+        secrets: |
+          ${{ inputs.nginx-plus == true && format('"nginx-repo.crt={0}"', secrets.NGINX_CRT) || '' }}
+          ${{ inputs.nginx-plus == true && format('"nginx-repo.key={0}"', secrets.NGINX_KEY) || '' }}
+
+    - name: 'Setup dotenv file'
+      working-directory: ./tests/scripts
+      env:
+        GKE_PROJECT_ID: ${{ secrets.GCP_PROJECT_ID }}
+        SERVICE_ACCOUNT: ${{ secrets.GCP_SERVICE_ACCOUNT }}
+        NODES_SERVICE_ACCOUNT: ${{ secrets.GKE_NODES_SERVICE_ACCOUNT }}
+      run: |
+        echo "RESOURCE_NAME=nfr-tests-${{ github.run_id }}" >> vars.env
+        echo "TAG=${{ inputs.version }}-${{ github.run_id }}" >> vars.env
+        echo "PREFIX=gcr.io/${GKE_PROJECT_ID}/ngf-nfr/nginx-gateway-fabric" >> vars.env
+        echo "NGINX_PREFIX=gcr.io/${GKE_PROJECT_ID}/ngf-nfr/nginx-gateway-fabric/nginx" >> vars.env
+        echo "NGINX_PLUS_PREFIX=gcr.io/${GKE_PROJECT_ID}/ngf-nfr/nginx-gateway-fabric/nginxplus" >> vars.env
+        echo "GKE_CLUSTER_NAME=nfr-tests-${{ github.run_id }}" >> vars.env
+        echo "GKE_CLUSTER_ZONE=us-east1-b" >> vars.env
+        echo "GKE_CLUSTER_REGION=us-east1" >> vars.env
+        echo "GKE_PROJECT=${GKE_PROJECT_ID}" >> vars.env
```

Comment:
Is `vars.env` going to be accessible in the pipeline, resulting in these variables being visible?

Reply:
The variables are redacted in the logs as they originated from secrets. There's nothing too sensitive anyway, none of the env vars are credential related, and we are using SSO and short lived credentials as it is, and that's all handled by the auth step above. Technically if someone had access to the runner they could get the file, but if they made it that far, they'd have access to more sensitive info than this.

Reply:
Here's an example of log output with the redaction:

```| go test -v ./suite  -args --gateway-api-version=1.0.0  \
|       --gateway-api-prev-version=1.0.0  --image-tag=edge-1 --plus-enabled=false \
|       --ngf-image-repo=gcr.io/***/ngf-nfr/nginx-gateway-fabric --nginx-image-repo=gcr.io/***/ngf-nfr/nginx-gateway-fabric/nginx --pull-policy=Always \
|       --k8s-version=latest  --service-type=LoadBalancer --is-gke-internal-lb=true```

I've refactored this a bit anyway to use the secrets directly in the job and skip the intermediate env vars, it works fine.